crates/gossip/src/state.rs

```diff
@@ -0,0 +1,194 @@
+use crate::protocol::{
+    K2GossipAcceptMessage, K2GossipAgentsMessage, K2GossipNoDiffMessage,
+};
+use kitsune2_api::{AgentId, K2Error, K2Result, Url};
+use rand::RngCore;
+
+/// The state of a gossip round.
+#[derive(Debug)]
+pub(crate) struct GossipRoundState {
+    /// The agent id of the other party who is participating in this round.
+    pub session_with_peer: Url,
+
+    /// The time at which this round was initiated.
+    ///
+    /// This is used to apply a timeout to the round.
+    #[allow(dead_code)]
+    started_at: std::time::Instant,
+
+    /// The session id of this round.
+    ///
+    /// Must be randomly chosen and unique for each initiated round.
+    pub session_id: bytes::Bytes,
+
+    /// The current stage of the round.
+    ///
+    /// Store the current stage, so that the next stage can be validated.
+    pub stage: RoundStage,
+}
+
+impl GossipRoundState {
+    /// Create a new gossip round state.
+    pub(crate) fn new(
+        session_with_peer: Url,
+        our_agents: Vec<AgentId>,
+    ) -> Self {
+        let mut session_id = bytes::BytesMut::with_capacity(96);
```

Comment:
Why 96?

Reply:
Chosen to be 12 bytes, I thought that was enough to be reasonably unique. Should I change it to 16?

Reply:
The capacity is in bytes, not bits. you have 96 here, not 12